hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java

```diff
@@ -334,13 +335,15 @@ public Set<Map.Entry<K, V>> entrySet() {
     @Override
     public Set<K> keySet(Predicate predicate) {
         checkNotNull(predicate, "Predicate cannot be null!");
+        PredicateUtils.checkDoesNotContainPagingPredicate(predicate, "keySet");
 
         return scanAndGetResult(predicate, ResultType.KEY);
     }
 
     @Override
     public Collection<V> values(Predicate predicate) {
         checkNotNull(predicate, "Predicate cannot be null!");
+        PredicateUtils.checkDoesNotContainPagingPredicate(predicate, "values");
```

Comment:
why do we have this check? shouldn't this method be able to specify a paging predicate?